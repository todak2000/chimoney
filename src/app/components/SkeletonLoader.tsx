/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { MdOutlineStarPurple500 } from "react-icons/md";
import { v4 as uuidv4 } from "uuid";

import { cn } from "../lib/cn";

interface LoaderProps {
  count?: number;
  className?: string;
  type?: string;
}
const SkeletonLoader: React.FC<LoaderProps> = ({
  count = 5,
  className,
  type,
}) => {
  if (type === "grid") {
    return (
      <div
        data-testid="grid-wrapper"
        className="grid animate-pulse grid-cols-1 gap-1 bg-white md:grid-cols-5"
      >
        {[...Array(count)].map((_) => (
          <div
            key={uuidv4()}
            data-testid="grid"
            className={cn("w-full rounded bg-gray-200", className)}
          />
        ))}
      </div>
    );
  }
  if (type === "block") {
    return (
      <div
        data-testid="block-wrapper"
        className="grid animate-pulse grid-cols-1  h-full"
      >
        {[...Array(count)].map((_) => (
          <div
            key={uuidv4()}
            data-testid="block"
            className={cn("w-full rounded bg-gray-200", className)}
          />
        ))}
      </div>
    );
  }
  if (type === "star") {
    return (
      <div
        data-testid="star-wrapper"
        className="grid animate-pulse grid-cols-5"
      >
        {[...Array(count)].map((_) => (
          <MdOutlineStarPurple500
            key={uuidv4()}
            data-testid="star"
            className={cn("mx-2 text-xl text-gray-300", className)}
          />
        ))}
      </div>
    );
  }

  return null;
};

export default SkeletonLoader;
