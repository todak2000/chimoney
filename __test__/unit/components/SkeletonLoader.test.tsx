import { render, screen } from "@testing-library/react";

import SkeletonLoader from "@/app/components/SkeletonLoader";

describe("Skeleton Loader Component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<SkeletonLoader />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders grid type without crashing", () => {
    render(<SkeletonLoader type="grid" count={5} />);
    const gridWrapper = screen.getAllByTestId("grid-wrapper");
    expect(gridWrapper).toHaveLength(1);
    const grids = screen.getAllByTestId("grid");
    expect(grids).toHaveLength(5);
  });

  it("renders block type without crashing", () => {
    render(<SkeletonLoader type="block" count={5} />);
    const blockWrapper = screen.getAllByTestId("block-wrapper");
    expect(blockWrapper).toHaveLength(1);

    const blocks = screen.getAllByTestId("block");
    expect(blocks).toHaveLength(5);
  });

  it("renders star type without crashing", () => {
    render(<SkeletonLoader type="star" count={5} />);
    const starWrapper = screen.getAllByTestId("star-wrapper");
    expect(starWrapper).toHaveLength(1);

    const stars = screen.getAllByTestId("star");
    expect(stars).toHaveLength(5);
  });
});
