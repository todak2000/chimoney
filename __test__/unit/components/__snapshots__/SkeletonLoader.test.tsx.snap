// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton Loader Component matches snapshot 1`] = `<DocumentFragment />`;
